### PR TITLE
feat(SolidJS): custom styling and content

### PR DESCRIPTION
**Pull request by Mr0Bread**
Functionality to customise styling and content for SolidJS package

The argument for content and appearance callbacks is a bit different due to the fact that reactivity works differently in Solid.

Docs were appended with Solid related info

### CODE REVIEW

File: docs/src/pages/nextjs/appdir.mdx [hunk 1]
```diff
@@ -72,10 +72,10 @@ export const { GET, POST } = createNextRouteHandler({
 
 ### Creating The UploadThing Components (optional)
 
-Generating components lets you pass your generic `FileRouter` once and then
-have typesafe components everywhere, instead of having to pass the generic
-everytime you mount a component, but you can also import the components
-individually from `@uploadthing/react`.
+Generating components lets you pass your generic `FileRouter` once and then have
```
Comment (t3dotgg) [inline docs/src/pages/nextjs/appdir.mdx:75]: Did the line wrap spacing amount change?

File: packages/react/src/components/dropzone.tsx [hunk 2]
```diff
@@ -57,17 +57,17 @@ export function UploadDropzone<TRouter extends FileRouter>(
   const $props = props as unknown as UploadDropzoneProps<TRouter> & {
     // props not exposed on public type
     // Allow to set internal state for testing
-    __internal_state?: "readying" | "ready" | "uploading",
+    __internal_state?: "readying" | "ready" | "uploading";
```
Comment (t3dotgg) [inline packages/react/src/components/dropzone.tsx:60]: What triggered this change?

**Review by t3dotgg — COMMENTED** [review]
This PR seems to have a lot of unrelated changes, did something slip through review that is causing lint failures on main?

File: packages/react/src/components/dropzone.tsx (hunk 2 above)
Comment (Mr0Bread) [reply packages/react/src/components/dropzone.tsx:60]: pnpm format

**Comment by Mr0Bread** [comment]
@markflorkowski PR updated. Docs were appended. Could you please take a look?

**Review by markflorkowski — APPROVED** [review]

**Pull request closed without merging**